### PR TITLE
Added toggle to turn off routes

### PR DESCRIPTION
**Pull request by BrandonMurnane**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -177,6 +178,16 @@ function start(clusterProcessId = 0) { // eslint-disable-line max-statements
     .readFileSync(conf.api.swagger.doc, ENCODING);
   const swaggerDoc = yaml.safeLoad(swaggerFile);
 
+  if (featureToggles.isFeatureEnabled('hideRoutes')) {
+    for (let _path in swaggerDoc.paths) {
+      if (swaggerDoc.paths.hasOwnProperty(_path)) {
+        if (conf.hiddenRoutes.includes(_path.split('/')[ONE])) {
```
Comment (kriscfoster) [inline index.js]: @BrandonMurnane is conf.hiddenRoutes defined somewhere or does this resolve to an environment variable? I don't see is in this PR

File: package.json [hunk 2]
```diff
@@ -59,6 +59,8 @@
     "test-limiter": "npm run test-with-coverage -- tests/limiter/limiter.js",
     "test-main": "npm run test-with-coverage -- --recursive tests/api tests/clock tests/config tests/db tests/enforceToken tests/realtime tests/enableCache tests/logging",
     "test-travis": "npm run test && npm run coveralls",
+    "test-routes-enabled": "HIDE_ROUTES=true HIDDEN_ROUTES=aspects mocha -R dot --recursive tests/routeFlag/FlagEnabled",
```
Comment (kriscfoster) [inline package.json]: @BrandonMurnane these test commands aren't getting called as part of the travis build.. I think you can add the calls for these to the end of test-extra or something similar

Comment (BrandonMurnane) [reply package.json]: thanks i missed that

File: index.js (hunk 1 above)
Comment (kriscfoster) [reply index.js]: Ah yes I see it now, thanks

File: tests/routeFlag/FlagDisabled.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/api/v1/routeFlag/FlagDisabled.js
```
Comment (iamigo) [inline tests/routeFlag/FlagDisabled.js]: ordinary module names start with lower case (unless it is a class name)

**Comment by iamigo** [comment]
> Refocus@1.0.0 test-routes-enabled /home/travis/build/salesforce/refocus
> HIDE_ROUTES=true HIDDEN_ROUTES=aspects mocha -R dot --recursive tests/routeFlag/flagEnabled
Warning: Could not find any test files matching pattern: tests/routeFlag/flagEnabled

**Pull request merged**